projects/core/src/button/button.element.ts

```diff
@@ -83,34 +83,42 @@ export class CdsButton extends CdsBaseButton {
   @property({ type: String })
   loadingState: keyof typeof ClrLoadingState = ClrLoadingState.default;
 
+  get disabled(): boolean {
+    return super.disabled;
+  }
+
+  set disabled(value: boolean) {
+    // track disabled state set from outside, in order to restore value after button is back in default loading state
+    this._disabled_from_outside = value;
+
+    // do NOT propagate disabled state, unless we're in default loading state
+    // otherwise button can be enabled while being in a non-default loading state, e.g. loading
+    if (this.isDefaultLoadingState(this.loadingState)) {
```

Comment:
I was playing around with my own implementation loosely based off of yours and I was able to get around having this conditional logic here. I think we can always call `super` here and it still more or less works the same.